libs/libvqm/vqm_common.c

```diff
@@ -360,7 +361,7 @@ t_pin_def *add_pin(char *name, int left, int right, t_pin_def_type type, t_parse
 
     //Append the element, the position is one less thatn the size of the array
     //which is returned by append_array_element
-    size_t position = append_array_element( (long) pin, pin_list) - 1;
+    size_t position = append_array_element( (intptr_t) pin, pin_list) - 1;
 
```

Comment:
I don't think you should change long to intptr_t.  We're using id-based structures in the code I looked at below, where we are inserting things into vectors, setting integer ids and such.  We don't need a pointer for that; just an integer of sufficient size. So long is appropriate for most cases, and for some smaller ids we use shorts.

Reply:
`long` => `int64_t` then? The problem with `long` on Windows is that it's not the same size as a pointer on 64-bit windows (`long` is 32 bit). So all the conversions from pointer to an integer will truncate (and cause compiler errors). What about a typedef such as `id_t` to convey meaning?

As for the compiler errors, I'll take a look. At least for the one you showed, looks like I forgot a header...

Reply:
@vaughnbetz Friendly ping... what should I do instead of changing `long` to `intptr_t`? Keeping `long` here will _not_ work because `long` is not 64-bits long, even on 64-bit Windows. So I need to change the type to something that will be the same size as a pointer (since we cast between pointer and int) on all platforms `vpr` runs on. Unless you prefer a platform-specific `typedef`, `intptr_t` seems to be the most appropriate type.